### ISSUE TITLE
select() obscure warning

### ISSUE THREAD

**Issue by lars20070** [Bioconductor/UniProt.ws#2]
```
a <- UniProt.ws::select(up, keys, columns, keytype)
Getting mapping data for P52875 ... and KEGG_ID
Getting extra data for P52875,Q9DCT2,Q9CR16,Q3U0V1,P62737,P23198
Getting extra data for O88685,Q5DTX6,C0HKE5,P60710,Q8VDI7,Q5SF07
Getting extra data for P27773,P60867,Q80WJ1,Q9JIL5,Q3UTZ3,Q99PI4
Getting extra data for Q9CQG2,Q8K2J7,Q9CZJ2,Q8CFE3,Q6P2K6,Q0VAY3
Getting extra data for Q9QZZ4,Q9CYA6,E9Q6B2,Q9CQ92,Q9D4I2,Q5S003
Getting extra data for Q9ER60,P97496,P23506,Q9D0L8,Q8R3R8,Q7TPD0
Getting extra data for Q9QUK4,P0C0S6,Q99LN9,Q9D5A0,P04247,Q923G2
'select()' returned many:many mapping between keys and columns
Warning message:
In matrix(vec2, ncol = 2, byrow = TRUE) :
  data length [377] is not a sub-multiple or multiple of the number of rows [189]
```
What does this warning message mean? Would it be possible to write a clearer exception?

**Comment by mtmorgan** [comment]
Can you provide a simple reproducible example, that I can copy into my R session and see the problem? Currently I have (not surprising, but to illustrate what I'm asking for!)

```
> a <- UniProt.ws::select(up, keys, columns, keytype)
Error in UniProt.ws::select(up, keys, columns, keytype) : 
  object 'up' not found
```

**Comment by lars20070** [comment]
```
library(UniProt.ws)

species <- 10090
columns <- c("SEQUENCE","GO", "SUBCELLULAR-LOCATIONS", "PROTEIN-NAMES", "GENES", "KEGG")

up <- UniProt.ws(taxId=species)

keytype <- "UNIPROTKB"
a <- UniProt.ws::select(up, keys, columns, keytype)
```
The `keys` vector you can find in [`Debug_20180509.RData`](https://www.dropbox.com/s/yob9ir9bwmm6o4o/Debug_20180509.RData?dl=1).

**Comment by lars20070** [comment]
Very sorry. Cannot reproduce the warning.
If I come across the same warning again, I will file a more careful issue with a minimal example.
Thanks for the very quick reply.

**Comment by mtmorgan** [comment]
When I run your code using 
```
> packageVersion("UniProt.ws")
[1] '2.21.0'
```
I get 
```
> a <- UniProt.ws::select(up, keys, columns, keytype)
Getting mapping data for P52875 ... and KEGG_ID
Getting extra data for P52875,Q9DCT2,Q9CR16,Q3U0V1,P62737,P23198
Getting extra data for O88685,Q5DTX6,C0HKE5,P60710,Q8VDI7,Q5SF07
Getting extra data for P27773,P60867,Q80WJ1,Q9JIL5,Q3UTZ3,Q99PI4
Getting extra data for Q9CQG2,Q8K2J7,Q9CZJ2,Q8CFE3,Q6P2K6,Q0VAY3
Getting extra data for Q9QZZ4,Q9CYA6,E9Q6B2,Q9CQ92,Q9D4I2,Q5S003
Getting extra data for Q9ER60,P97496,P23506,Q9D0L8,Q8R3R8,Q7TPD0
Getting extra data for Q9QUK4,P0C0S6,Q99LN9,Q9D5A0,P04247,Q923G2
'select()' returned many:many mapping between keys and columns
Warning messages:
1: In matrix(vec2, ncol = 2, byrow = TRUE) :
  data length [377] is not a sub-multiple or multiple of the number of rows [189]
2: In file(file, "rt") :
  URL 'http://www.uniprot.org/uniprot/?query=O88685+or+Q5DTX6+or+C0HKE5+or+P60710+or+Q8VDI7+or+Q5SF07+or+Q8CGM1+or+Q8BM39+or+Q9R233+or+Q8BSU7+or+Q3TTY5+or+Q3V009+or+P10854+or+Q9CXE7+or+Q91VM5+or+Q8VEJ9+or+P63268+or+Q9WVA3+or+Q61171+or+P45591+or+P14211+or+Q9R0P6+or+P62082+or+Q9R0I7+or+Q60864+or+Q9D4B1+or+Q91ZC0+or+Q61070+or+P26039+or+Q9D211+or+O70194+or+Q99K01+or+Q9JII5+or+Q3UFY7+or+Q9CYZ6+or+Q9EPK7+or+P62322+or+Q8R429+or+Q9D1A2+or+Q60902+or+Q6V5K9+or+Q8VHL1+or+P26443+or+Q80YV2+or+Q7TNE3+or+Q9DCL9+or+P19096+or+Q6Y5D8+or+O35239+or+O88983+or+P35505+or+Q61140+or+B2RXR6+or+Q8BGS0+or+Q75N73+or+Q3TXS7+or+P35761+or+Q8BFY9+or+P63325+or+Q9DCD5+or+Q6P9P6+or+Q8CFE6+or+P11214+or+P17751+or+P47911+or+P63254+or+Q9D5T0+or+Q8VHP6+or+O08638+or+P60762+or+P56959+or+Q9CWR0+or+Q80X19+or+P62264+or+P11983+or+P83940+or+Q9JHJ0+or+Q9JHI7+or+Q52KI8+or+Q61205+or+Q6ZWV7+or+P47199+or+Q8C3B8+or+Q9CX86+or+Q9JKF1+or+P47878+or+P62317+or+O89086+or+O35250+or+Q8CG47+or+Q8CG46+or+P62270+or+Q99K48+or+Q61733+or+P56695+or+Q8C9M2+ [... truncated]
3: In file(file, "rt") :
  cannot open URL 'http://www.uniprot.org/uniprot/?query=Q9QZZ4+or+Q9CYA6+or+E9Q6B2+or+Q9CQ92+or+Q9D4I2+or+Q5S003+or+Q58A65+or+P35821+or+Q91WN1+or+P0CG09+or+O35127+or+Q9DBS2+or+P97434+or+Q8BFR5+or+Q6DFV7+or+Q8CDS7+or+Q6XP49+or+Q8VC19+or+Q9R001+or+P39749+or+P83882+or+Q5F2E7+or+Q9Z2T6+or+Q9Z120+or+Q9JKP8+or+P97767+or+Q6P6L0+or+Q9WV30+or+Q9CYK1+or+Q9CQ06+or+Q6DG52+or+Q8R4C2+or+Q3UZ39+or+P97360+or+Q9CXV1+or+Q8CGE8+or+P61514+or+Q3V1V3+or+Q9JJ06+or+P54071+or+P62751+or+P28741+or+O88271+or+Q62376+or+Q9CQ65+or+O35309+or+P31001+or+P58871+or+P83870+or+Q64131+or+P56394+or+Q4VBF2+or+P54923+or+Q62009+or+P21981+or+Q8BMS1+or+Q9DCB8+or+Q9R0X4+or+Q921G7+or+Q924L1+or+P27659+or+Q6ZWN5+or+Q9R244+or+Q8QZV7+or+P53996+or+Q9D338+or+Q9JKF6+or+Q6NXY1+or+Q8C0L0+or+Q8R0A0+or+Q8BVL3+or+P81069+or+Q8CI75+or+Q60973+or+Q8BGD6+or+Q9D0S9+or+P18406+or+Q3U284+or+Q9EPL0+or+Q9ERS2+or+Q80U87+or+Q91ZW3+or+Q5SSG4+or+Q9D4H2+or+P34884+or+P19785+or+Q9Z1Y4+or+D3YYU8+or+Q9DB20+or+Q9CXI5+or+P34152+or+Q80ZK0+or+Q99MQ3+or+P14685+or+Q7TNV [... truncated]
```
Setting `options(warning.length = 8170)` and re-running reveals the full output, indicating that the URL is too long (R prints the long URL, then the message, so the message is truncated). I'm not sure why this is only reproducible in a transient way; it failed for me the first time but not the second time. The 'url too long' message is coming from the server, and perhaps UniProt has servers that are configured differently.

Basically, the error comes about because you are asking for too many keys.

**Comment by lars20070** [comment]
@mtmorgan Very helpful. Thank you.
We will split our tables into several smaller ones.
